src/Symfony/Component/Security/Http/Firewall/LogoutListener.php

```diff
@@ -34,31 +37,47 @@ class LogoutListener extends AbstractListener
 {
     private $tokenStorage;
     private $options;
-    private $handlers;
-    private $successHandler;
     private $httpUtils;
     private $csrfTokenManager;
+    private $eventDispatcher;
 
     /**
      * @param array $options An array of options to process a logout attempt
      */
-    public function __construct(TokenStorageInterface $tokenStorage, HttpUtils $httpUtils, LogoutSuccessHandlerInterface $successHandler, array $options = [], CsrfTokenManagerInterface $csrfTokenManager = null)
+    public function __construct(TokenStorageInterface $tokenStorage, HttpUtils $httpUtils, /* EventDispatcherInterface */$eventDispatcher, array $options = [], CsrfTokenManagerInterface $csrfTokenManager = null)
     {
+        if (!$eventDispatcher instanceof EventDispatcherInterface) {
+            trigger_deprecation('symfony/security-http', '5.1', 'Passing a logout success handler to "%s" is deprecated, pass an instance of "%s" instead.', __METHOD__, EventDispatcherInterface::class);
+
+            if (!$eventDispatcher instanceof LogoutSuccessHandlerInterface) {
+                throw new \InvalidArgumentException(sprintf('Argument 3 of "%s" must be instance of "%s" or "%s", "%s" given.', __METHOD__, EventDispatcherInterface::class, LogoutSuccessHandlerInterface::class, get_debug_type($successHandler)));
+            }
+
+            $successHandler = $eventDispatcher;
+            $eventDispatcher = new EventDispatcher();
+            $eventDispatcher->addListener(LogoutEvent::class, function (LogoutEvent $event) use ($successHandler) {
```

Comment:
we still use named events, don't we?

Also, the closure can be made static.

Reply:
IIRC we allowed listening both on named and FQCN events in core, right? Happy to change this to a named event tho.

Reply:
`DeauthenticatedEvent` is not named either. Seems consistent